src/plugins/explore/public/application/utils/state_management/utils/get_prompt_mode_is_available/get_prompt_mode_is_available.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Dataset } from '../../../../../../../data/common';
+import { ExploreServices } from '../../../../../types';
+
+export const getPromptModeIsAvailable = async (
+  services: ExploreServices,
+  dataset: Dataset
+): Promise<boolean> => {
+  try {
+    const extensions = services.data.query.queryString
+      .getLanguageService()
+      .getQueryEditorExtensionMap();
+
+    // Check if query assist is available through data plugin extension system
+    if (!extensions['query-assist']) {
+      return false;
+    }
+
+    // check if agent is available
+    const response: { configuredLanguages: string[] } = await services.http.get(
+      '/api/enhancements/assist/languages',
+      {
+        query: { dataSourceId: dataset.dataSource?.id },
+      }
+    );
```

Comment:
can this use `isEnabled$` of the query assist extension? to avoid another network call